staker-rewards/lib/fraud-detector.js

```diff
@@ -0,0 +1,47 @@
+import clc from "cli-color"
+
+export default class FraudDetector {
+  constructor(context, tokenStaking, factoryDeploymentBlock) {
+    this.context = context
+
+    this.tokenStaking = tokenStaking
+
+    this.factoryDeploymentBlock = factoryDeploymentBlock
+  }
+
+  static async initialize(context) {
+    const { TokenStaking, factoryDeploymentBlock } = context.contracts
+
+    const tokenStaking = await TokenStaking.deployed()
+
+    return new FraudDetector(context, tokenStaking, factoryDeploymentBlock)
+  }
+
+  async isOperatorFraudulent(operator) {
+    // Get all slashing events for the operator.
```

Comment:
Could be worth adding a log:
```
    console.log(`Checking fraudulent activity for operator ${operator}`)
```
so that there is no effect of the script hanging. We log `Checking current status of keep 0x...` and `Checking signature requests for keep 0x` before.